### PR TITLE
Add metrics to core API

### PR DESCRIPTION
**Pull request by josephmckinsey**
Implements the metrics for mean absolute relative (percentage) error, mean absolute error for angles, and root mean squared relative error.

Unexpected difficulties:
- units from PNNL's DSSE was in degrees
- DSSE timing does not quite match the feeder simulator. I use the intersection of the rows, which is somewhat pessimistic especially if the DSSE reports more often.

### CODE REVIEW

File: oedisi/tools/metrics.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    if angle_unit == "degrees":  # convert to radians
+        estimated_angle = estimated_angle * np.pi / 180
+
+    # Convert true voltages to magnitude and angles
+    true_magnitudes = np.hypot(true_voltages_real, true_voltages_imag)
+    true_angles = np.arctan2(true_voltages_imag, true_voltages_real)
+
+    nonzero_parts = true_magnitudes != 0
+    assert (
+        true_magnitudes.columns == estimated_magnitude.columns
```
Comment (tarekelgindy) [inline oedisi/tools/metrics.py:126]: question: should this use an isclose() type of computation? Or is it ok to evaluate the values exactly?

File: tests/unit_tests/test_oedisi_tools/test_metrics.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+def test_metrics(base_path: Path, monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.chdir(base_path)
+
+    runner = CliRunner()
+    result = runner.invoke(
+        cli, ["evaluate-estimate", "--path", Path("test_data"), "--metric", "MARE"]
+    )
+    assert result.exit_code == 0
+    assert np.isclose(float(result.output), 0.6281941853348966)
```
Comment (tarekelgindy) [inline tests/unit_tests/test_oedisi_tools/test_metrics.py:22]: Just checking - where are these hardcoded values taken from? Might there be differences based on the performance of the algorithm?

**Review by tarekelgindy — APPROVED** [review]
I added a couple of comments that may want to be addressed, but for the most part I think it looks fine.

File: tests/unit_tests/test_oedisi_tools/test_metrics.py (hunk 2 above)
Comment (josephmckinsey) [reply tests/unit_tests/test_oedisi_tools/test_metrics.py:22]: This is running against the test data, which is very fixed.

**Pull request merged**